lib/chef/knife/bmcs_server_create.rb

```diff
@@ -66,6 +66,19 @@ class BmcsServerCreate < Knife
                           "is a convenience wrapper around the 'ssh_authorized_keys' field of the --metadata parameter. Populating both values in the same call will result "\
                           'in an error. For more info see documentation: https://docs.us-phoenix-1.oraclecloud.com/api/#/en/iaas/20160918/requests/LaunchInstanceDetails.'
 
+      option :assign_public_ip,
+             long: '--assign-public-ip BOOLEAN',
```

Comment:
Good point - I see that that boolean flag doesn't work the way I thought it did.